include/motor.hpp

```diff
@@ -0,0 +1,99 @@
+#ifndef MOTOR_H
+#define MOTOR_H
+
+#include "freertos/FreeRTOS.h"
+#include "freertos/task.h"
+#include "esp_log.h"
+//#include "freertos/semphr.h"
+
+/**
+ *  Motor
+ *   
+ *  position = mm
+ *  speed = mm/s
+ *  acceleration = mm/s^2
+ */
+
+#define DEBUG_CLIPPING              // Show debug messages when motions violating the machine 
+                                    // physics are commanded
+
+// If start > end, indicates an inversion of direction
+typedef struct {
```

Comment:
Think about how bounds are being handled harder. There is some wierdness when start is not 0 in the current code

Reply:
Yeah, I definitely want the internal coordinate system StrokeEngine uses to be what you described. I think I was thinking for case where you want the stroke bounds to be some offset position for linmots, or an offset home. I originally had to deal with 110mm being home on the linmots due to some misconfig on my part, so I think that's where this came from.

I bet I can improve this though, I just need to think on it a bit, and make sure I've got it all right in my head

Reply:
If there is such a requirement this should maybe be part of the motor_driver.h itself. It may have a different coordinate system internally, but expose the convention to the outside world.

Reply:
Do we even need to specify `start` if it's always 0? Do we just need a length/keepout property? I think that's what you are saying with only `physicalTravel and keepoutBoundary` are needed comment.

Reply:
We will be keeping start